File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<int>chk;
        int l=0;
        int r=0;
        int maxi = 0;
        int n = s.length();
        while(r<n){
            char c = s[r];
            while(chk.find(c)!=chk.end()){
                chk.erase(s[l]);
                l++;
            } 
            chk.insert(s[r]);
            maxi = max(maxi,r-l+1);
            r++;

            }

        return maxi;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        chari = set()
        l = 0 
        ans =0
        n = len(s)
        for ch in range(n):
            while s[ch] in chari:
                chari.remove(s[l])
                l+=1
            chari.add(s[ch])
            ans = max(ans,ch-l+1)
        return ans